src/components/ContentPreview/ContentPreview.js

```diff
@@ -409,7 +417,8 @@ class ContentPreview extends PureComponent<Props, State> {
             useHotkeys: false
         };
 
-        this.preview = new Preview();
+        // Use instance of Preview from props if provided
+        this.preview = previewInstance || new Preview();
```

Comment:
Don't think it should be here, because you only want it set the 1st initial time. Every other time it should create a new instance when things like navigation happen.

Reply:
I think you may have to extract out the lines below, including listeners, in its own function like `createPreviewInstance(preview) { ... }` where it can be called from the constructor if it has been primed already.